singularity-eos/eos/eos.hpp

```diff
@@ -154,6 +326,146 @@ class Gruneisen {
   void ValuesAtReferenceState(Real &rho, Real &temp, Real &sie, Real &press, Real &cv,
                               Real &bmod, Real &dpde, Real &dvdt,
                               Real *lambda = nullptr) const;
+
+  /*
+  Vector versions of the member functions are only available for CPU
+  
+  RealIndexer must have an operator[](int) that returns a Real. e.g., Real*
+  ConstRealIndexer is as RealIndexer, but assumed const type.
+  LambdaIndexer must have an operator[](int) that returns a Real*. e.g., Real**
```

Comment:
perhaps worth commenting that unlike the scalar versions of the functions, the lambda indexer is not optional here.

We could make it optional by adding an overload with the lambdas missing and a dummy lambda indexer passed in perhaps.

Reply:
Good point. I'd prefer to mirror the functionality of the scalar functions as much as possible so I'll overload the function